packages/web-client/app/models/workflow/postable-collection.ts

```diff
@@ -49,6 +49,13 @@ export default class PostableCollection {
       }
 
       if (!post.isComplete) {
+        // clear out remaining posts from #includedPostables and #excludedPostables,
+        // in case we have a situation where we went backward
+        for (let post of this.postables.slice(i, -1)) {
```

Comment:
`Array.slice(i, -1)` always leaves out the last item of an array. This means that a milestone that has one `WorkflowCard`, A, as its last item, and another, B, before it, will have A automatically marked as completed if B's `onIncomplete` method is called and then B is completed. For this workflow, I think this doesn't affect anything, but implementation might need to change for the 'issue card' workflow.

Reply:
Example here: https://github.com/cardstack/cardstack/tree/example/incompletion-bug

Reply:
Good catch. I pushed an update with an improved approach.

Reply:
Thanks @lukemelia, it works nicely! I've added a test for the Workflow calling Milestone `resetFrom` with the right arguments. Let's merge if [this commit](https://github.com/cardstack/cardstack/pull/1574/commits/1f6e78a384942beeca73161c2ea269b49bdf5c67) looks ok?

Reply:
added some detail to the epilogue test.